src/interpreter/executors/main.rs

```diff
@@ -135,7 +135,11 @@ where
                 // TODO set tree balance
 
                 let program = self.run_program();
-                let gas_used = self.transaction().gas_limit() - self.registers[REG_GGAS];
+                let gas_used = self
+                    .transaction()
+                    .gas_limit()
+                    .checked_sub(self.registers[REG_GGAS])
+                    .ok_or(InterpreterError::Panic(PanicReason::ArithmeticOverflow))?;
```

Comment:
Why is the panic reason `ArithmeticOverflow`? It doesn't feel right as this is a gas-related issue

Reply:
True. This error should not happen in normal cases for sound transactions, so its debatable on whether we should use canonical errors for this exception, or just classify it by what it is: an underflow.

But then it might be better to stick to the easier to track option that is out of gas: https://github.com/FuelLabs/fuel-vm/pull/184/commits/9cf81732a0ac3c2b96932858bf86d69a79223f37